src/Tasks/Microsoft.NET.Build.Tasks/build/Microsoft.NET.Sdk.Common.targets

```diff
@@ -76,6 +76,14 @@ Copyright (c) .NET Foundation. All rights reserved.
                                Condition="'$(_SkipNearestTargetFrameworkResolution)' != 'true'">
       <Output PropertyName="NearestTargetFramework" TaskParameter="NearestTargetFramework" />
     </GetNearestTargetFramework>
+
+    <ItemGroup>
+      <ProjectBuildInstructions Include="$(MSBuildProjectFullPath)">
+        <DesiredTargetFrameworkProperties>TargetFramework=$(NearestTargetFramework)</DesiredTargetFrameworkProperties>
+        <HasSingleTargetFramework>$(_HasSingleTargetFramework)</HasSingleTargetFramework>
```

Comment:
Ok, that's possible, but I don't think I can get through implementing it today.

It's easy enough to make `GetDesiredProperties` return the right set of properties to set and to remove.

Where it's hard is removing the right properties for projects that want to short-circuit by saying "I don't cross-target, just strip the properties"--and not reevaluate with ReferringTF. We could change the protocol on `ShouldQueryForProperties` to instead get back an item with the project name and metadata for removed properties, basically the same way `GetDesiredProperties` does now. That's a harder thing to implement (especially in the default case) but not too bad.

@AndyGerlicher does that description make enough sense for you to pick this up?